pkg/tlsconfig/tlsconfig.go

```diff
@@ -0,0 +1,55 @@
+// Package tlsconfig provides helper to load a robust TLS config.
+package tlsconfig
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+)
+
+// LoadTLSConfig loads the certs from file.
+//
+// NOTE: We currently use the same CA for clients/server, for production, should use distinct ones.
+func LoadTLSConfig(certFile, keyFile, caFile string, isClient bool) (*tls.Config, error) {
+	certificate, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load certificate: %w", err)
+	}
+
+	ca, err := os.ReadFile(caFile)
+	if err != nil {
+		return nil, fmt.Errorf("faild to read CA certificate: %w", err)
+	}
+
+	capool := x509.NewCertPool()
+	if !capool.AppendCertsFromPEM(ca) {
+		return nil, errors.New("unable to append the CA certificate to CA pool") //nolint: err113 // Acceptable.
+	}
+
+	tlsConfig := &tls.Config{
+		MinVersion: tls.VersionTLS13,
+		CurvePreferences: []tls.CurveID{
+			tls.X25519,
+			tls.CurveP521,
+			tls.CurveP384,
+			tls.CurveP256,
+		},
+		CipherSuites: []uint16{
+			tls.TLS_CHACHA20_POLY1305_SHA256,
+			tls.TLS_AES_256_GCM_SHA384,
+			tls.TLS_AES_128_GCM_SHA256,
+		},
```

Comment:
This doesn't do anything, ciphersuites are ignored in TLS 1.3 (and I suspect that the awkwardness of expressing an order based on the availability of AES acceleration is one of the reasons).

Reply:
Indeed, I added to for info. I’ll remove it.